docker/sgx.dockerfile

```diff
@@ -1,16 +1,46 @@
 FROM ubuntu:20.04
 
+ENV DEBIAN_FRONTEND=noninteractive
+
 # System deps
 RUN apt-get update
-RUN apt-get install -y \
-    dkms \
-    nasm \
-    wget
+RUN apt-get install  -y \
+    autoconf \
+    automake \
+    build-essential \
+    cmake \
+    curl \
+    debhelper \
+    git \
+    libcurl4-openssl-dev \
+    libprotobuf-dev \
+    libssl-dev \
+    libtool \
+    lsb-release \
+    ocaml \
+    ocamlbuild \
+    protobuf-compiler \
```

Comment:
Do we really need all these packages just to install SGX e.g. `protobuf-compiler` and OCaml? Seems like a lot.

Reply:
Yes, we are building and installing the code for both the SDK and PSW. The dependency list is a combination of:
* The dependencies listed in the [Pre-requisites](https://github.com/intel/linux-sgx#prerequisites)
* The [docker image](https://github.com/intel/linux-sgx/blob/f607f89ab1ac648c3e62a23991befdeb93f9bf9d/docker/build/Dockerfile#L33-L50) they provide (note that they use `18:04` as base, what changes some packages).

Reply:
Ok wow, I didn't expect it to require so much stuff. Would be good to link those in a comment to explain why it's such a big list.

Reply:
I add one comment to `sgx.dockerfile`.